### PR TITLE
Document more differences between SafeDI and manual DI

### PR DESCRIPTION
**Pull request by dfed**
We've discovered a few more key differences between manual DI and SafeDI, so we're now documenting them here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -407,6 +407,18 @@ The `@StateObject` documentation reads:
 
 `@Instantiated`, `@Forwarded`, or `@Received` objects may be decorated with [`@ObservedObject`](https://developer.apple.com/documentation/swiftui/ObservedObject). Note that `@Instantiated` objects declared on a `View` will be re-initialized when the view is re-initialized. You can find a deep dive on SwiftUI view lifecycles [here](https://www.donnywals.com/understanding-how-and-when-swiftui-decides-to-redraw-views/).
 
+#### Dependency cycles
+
+In a manual DI system it is easy to instantiate an object that can instantiate another object that can then instantiate the first object again. Instantiable objects in SafeDI, however, are not able to generate this kind of circular dependency. Instantiable objects can achieve this kind of circular dependency manually: one object in the cycle must call other cycle objects‘ memberwise initializers directly, rather than utilizing generated `Instantiator`s or `ForwardingInstantiator`s.
```
Comment (dfed) [inline README.md:412]: imo dependency cycles should be avoided... but it's important to call out what is required to enable their creation for teams that need this.

File: README.md [hunk 2]
```diff
@@ -407,6 +407,18 @@ The `@StateObject` documentation reads:
...
+
+#### Inheritance
+
+In a manual DI system it is simple for superclasses to receive injected dependencies. SafeDI‘s utilization of macros means that SafeDI is not aware of dependencies required due to inheritance trees. Due to this limitation, superclass types should not be decorated with `@Instantiable`: instead, subclasses should declare the properties their superclasses need, and pass them upwards via a call to `super.init(…)`.
+
+#### Nested types
+
+While manually written dependency injection code can work seamlessly with [nested types](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/nestedtypes/), SafeDI currently prevents nested types from being decorated with the `@Instantiable` macro.
```
Comment (dfed) [inline README.md:420]: these two limitations were well-known when this library was designed – I had forgotten to document them.

File: README.md [hunk 3]
```diff
@@ -407,6 +407,18 @@ The `@StateObject` documentation reads:
 
 `@Instantiated`, `@Forwarded`, or `@Received` objects may be decorated with [`@ObservedObject`](https://developer.apple.com/documentation/swiftui/ObservedObject). Note that `@Instantiated` objects declared on a `View` will be re-initialized when the view is re-initialized. You can find a deep dive on SwiftUI view lifecycles [here](https://www.donnywals.com/understanding-how-and-when-swiftui-decides-to-redraw-views/).
 
+#### Dependency cycles
+
+In a manual DI system it is easy to instantiate an object that can instantiate another object that can then instantiate the first object again. Instantiable objects in SafeDI, however, are not able to generate this kind of circular dependency. Instantiable objects can achieve this kind of circular dependency manually: one object in the cycle must call other cycle objects‘ memberwise initializers directly, rather than utilizing generated `Instantiator`s or `ForwardingInstantiator`s.
+
+#### Inheritance
```
Comment (dfed) [inline README.md:414]: while I'd love to tell people "just use composition"... SafeDI advertises itself as architecture independent. So instead we explain how to utilize SafeDI with inheritance trees.

File: README.md (hunk 1 above)
Comment (dfed) [reply README.md:412]: I think I have a way to fix this. But it's going to take time I don't have at the moment. In the interim, we should document this shortcoming.

File: README.md [hunk 4]
```diff
@@ -407,6 +407,18 @@ The `@StateObject` documentation reads:
 
 `@Instantiated`, `@Forwarded`, or `@Received` objects may be decorated with [`@ObservedObject`](https://developer.apple.com/documentation/swiftui/ObservedObject). Note that `@Instantiated` objects declared on a `View` will be re-initialized when the view is re-initialized. You can find a deep dive on SwiftUI view lifecycles [here](https://www.donnywals.com/understanding-how-and-when-swiftui-decides-to-redraw-views/).
 
+#### Dependency cycles
+
+In a manual DI system it is easy to instantiate an object that can instantiate another object that can then instantiate the first object again. Instantiable objects in SafeDI, however, are not able to generate this kind of circular dependency. Instantiable objects can achieve this kind of circular dependency manually: one object in the cycle must call other cycle objects‘ memberwise initializers directly, rather than utilizing generated `Instantiator`s or `ForwardingInstantiator`s.
+
```
Comment (dfed) [inline README.md:413]: Removing this new section entirely. We'll fix this with #50.
```suggestion
```

File: README.md [hunk 5]
```diff
@@ -425,7 +433,7 @@ To install the SafeDI framework into your package with [Swift Package Manager](h
 
 ```swift
 dependencies: [
-    .package(url: "https://github.com/dfed/SafeDI", from: "0.2.0"),
+    .package(url: "https://github.com/dfed/SafeDI", from: "0.3.0"),
```
Comment (dfed) [inline README.md:436]: bumped this while I was updating the README

**Review by MrAdamBoyd — APPROVED** [review]

**Pull request merged**